### PR TITLE
Core/Spell: Mage Ice/Prismatic/Blazing Barrier

### LINKED ISSUE #19110: [master] Mage - Ice Barrier
[//]: # (***************************************************************)
[//]: # (** DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED **)
[//]: # (***************************************************************)

**Description:**
Ice Barrier is outdated.

- http://www.wowhead.com/spell=11426/ice-barrier

**Current behaviour:** 
Ice Barrier is outdated.

**Expected behaviour:**
Ice Barrier should updated to 7.1.5

**Steps to reproduce the problem:**

1. Select Mage
2. Level up to 26
3. Change spec to Frost

**Branch(es):** master

**TC rev. hash/commit:**  https://github.com/TrinityCore/TrinityCore/commit/8aacc07727835ad3f81bebde3308cfdc4de2f26e

**TDB version:**  703.00 + updates

**Operating system:**  Windows 10 64-bit

[//]: # (This template is for problem reports. For other types of report, edit it accordingly.)
[//]: # (If this is a crash report, include the crashlog with https://gist.github.com/)
[//]: # (For fixes containing C++, create a Pull Request.)

### PR DESCRIPTION
**Pull request by DoctorKraft**
**Changes proposed:**

-  Update [Ice Barrier](http://www.wowhead.com/spell=11426/ice-barrier) (new calculation for absorb & add proc [Chilled](http://www.wowhead.com/spell=205708/chilled))
-  Fix [Blazing Barrier](http://www.wowhead.com/spell=235313/blazing-barrier) (add calculation for absorb & add proc [Blazing Barrier Trigger](http://www.wowhead.com/spell=235314/blazing-barrier))
-  Fix [Prismatic Barrier](http://www.wowhead.com/spell=235450/prismatic-barrier) (add calculation for absorb)

**Target branch(es):**

- [X] master

**Issues addressed:** Closes #19110 #19088 

**Tests performed:** Build & Tested in-game

### CODE REVIEW

**Comment by matukaa** [comment]
Umm, Prismatic barrier now uses blazing barrier's script, that's not okay. Blazing Barrier has an effect that is not present for prismatic barrier (will spam console with error)

Check the spellid in Register(), and bind that effect only to blazing barrier

**Comment by DoctorKraft** [comment]
@matukaa 

> Prismatic barrier now uses blazing barrier's script, that's not okay.

 what ? I don't understand, i have separate script. With artifact talent & honor talent after we need recalculate the amount of absorb

**Comment by matukaa** [comment]
Nvm, sorry, my mistake, seemingly I cant read

**Comment by Temppus** [comment]
/home/travis/build/TrinityCore/TrinityCore/src/server/scripts/Spells/spell_mage.cpp:673:43: fatal error: 
      unused parameter 'spellInfo' [-Wunused-parameter]
           bool Validate(SpellInfo const* spellInfo) override

File: src/server/scripts/Spells/spell_mage.cpp [hunk 1]
```diff
@@ -232,6 +228,54 @@ class spell_mage_blast_wave : public SpellScriptLoader
...
+                return ValidateSpellInfo
+                ({
+                    SPELL_MAGE_BLAZING_BARRIER_TRIGGER
+                });
+            }
+
+            void CalculateAmount(AuraEffect const* /*aurEff*/, int32& amount, bool& canBeRecalculated)
+            {
+                canBeRecalculated = false;
+                if (Unit* caster = GetCaster())
```
Comment (GitVerge) [inline src/server/scripts/Spells/spell_mage.cpp]: no need to check this

Comment (Keader) [inline src/server/scripts/Spells/spell_mage.cpp]: caster can be null in aura script, check should stay

Comment (Ascathor) [inline src/server/scripts/Spells/spell_mage.cpp]: Me no agree. Leave it there.

Comment (GitVerge) [inline src/server/scripts/Spells/spell_mage.cpp]: oh is aurascript nvm lol

**Pull request closed without merging**